### PR TITLE
Search engine init project setup

### PR DESCRIPTION
**Pull request by arezookmn**

### CODE REVIEW

**Comment by thisisnabi** [comment]
Please follow the `mode` and use layered architecture for this project

File: src/services/search-engine/Twitter.Clone.Search.Presentation/Program.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        app.UseHttpsRedirection();
+
+        app.UseAuthorization();
+
+
+        app.MapControllers();
+
+        app.Run();
+    }
+}
```
Comment (Javidleo) [inline src/services/search-engine/Twitter.Clone.Search.Presentation/Program.cs:35]: Use Top Level Statement Strcuture is profred for .NET projects (above .NET 5).

File: src/services/search-engine/Twitter.Clone.Search.Presentation/Controllers/BaseController.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace Twitter.Clone.Search.Presentation.Controllers;
+
+[Route("[controller]/[action]"), ApiController]
+public abstract class BaseController : ControllerBase
+{
+    private ISender? _mediator;
+
+    protected ISender Mediator => _mediator ??= HttpContext.RequestServices.GetRequiredService<ISender>();
+}
```
Comment (Javidleo) [inline src/services/search-engine/Twitter.Clone.Search.Presentation/Controllers/BaseController.cs:12]: - the base route for Controllers should be "api/[controller]" action part is unecessary.
- the _mediator should be a type of 'IMediator' not ISender.
- mediator should be injected via constructor not the Service Provider.

**Review by Javidleo — CHANGES_REQUESTED** [review]
fix the problems mentioned in the comments and return it back to me.
generally it was perfect, good job!

File: src/services/search-engine/Twitter.Clone.Search.Presentation/Controllers/BaseController.cs (hunk 2 above)
Comment (thisisnabi) [reply src/services/search-engine/Twitter.Clone.Search.Presentation/Controllers/BaseController.cs:12]: That is incorrect @Javidleo , you can use ISender when you want use Send role in mediatR package.

Comment (thisisnabi) [reply src/services/search-engine/Twitter.Clone.Search.Presentation/Controllers/BaseController.cs:12]: Select good conversion name for it arezoo such as `_sender`

Comment (thisisnabi) [reply src/services/search-engine/Twitter.Clone.Search.Presentation/Controllers/BaseController.cs:12]: @Javidleo There was no need to use these packages in the initial structure of the project.
Avoid useless things and don't look for `mode`.

File: src/services/search-engine/Twitter.Clone.Search.Presentation/Program.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+
+    var builder = WebApplication.CreateBuilder(args);
+
+    // Add services to the container.
+
+    builder.Services.AddControllers();
```
Comment (Javidleo) [inline src/services/search-engine/Twitter.Clone.Search.Presentation/Program.cs:6]: Move these services to presentation service registration, 
and call the registration methods you provided for each layer in this structure.

File: src/services/search-engine/Twitter.Clone.Search.Presentation/Program.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+    builder.Services.AddEndpointsApiExplorer();
+    builder.Services.AddSwaggerGen();
```
Comment (Javidleo) [inline src/services/search-engine/Twitter.Clone.Search.Presentation/Program.cs:9]: Add Presentation services to its own Service Registry method. 
AddControllers()
AddEndpointsApiExplorer()
AddSwaggerGen() 
and after that call Service Registry methods here.

**Review by Javidleo — CHANGES_REQUESTED** [review]
still need changes,

for next time you fixed the problems, change the label to `ready to review`

File: src/services/search-engine/Twitter.Clone.Search.Presentation/Controllers/BaseController.cs (hunk 2 above)
Comment (Javidleo) [reply src/services/search-engine/Twitter.Clone.Search.Presentation/Controllers/BaseController.cs]: > @Javidleo There was no need to use these packages in the initial structure of the project. Avoid useless things and don't look for `mode`.

yeah, you right. we'll try to keep it lightweight.

File: src/services/search-engine/Twitter.Clone.Search.Presentation/Program.cs (hunk 4 above)
Comment (arezookmn) [reply src/services/search-engine/Twitter.Clone.Search.Presentation/Program.cs]: Thank you for pointing that out, I've made the changes as suggested

**Review by Javidleo — APPROVED** [review]
done.

**Pull request merged**